sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -1353,14 +1396,44 @@ public int compare(Method o1, Method o2) {
   }
 
   /**
-   * A {@link Predicate} that returns true if the method is annotated with
-   * {@link JsonIgnore @JsonIgnore}.
+   * A {@link Predicate} that returns true if the method is annotated with {@code annotationClass}.
    */
-  static class JsonIgnorePredicate implements Predicate<Method> {
-    static final JsonIgnorePredicate INSTANCE = new JsonIgnorePredicate();
+  static class AnnotationPredicate implements Predicate<Method> {
+    static final AnnotationPredicate JSON_IGNORE = new AnnotationPredicate(
+        new Predicate<Method>() {
+          @Override
+          public boolean apply(@Nonnull Method input) {
+            return input.isAnnotationPresent(JsonIgnore.class);
+          }
+        },
+        JsonIgnore.class);
+
+    static final AnnotationPredicate DEFAULT_VALUE = new AnnotationPredicate(
+        new Predicate<Method>() {
+          @Override
+          public boolean apply(@Nonnull Method input) {
+            for (Class<?> klass : Default.class.getDeclaredClasses()) {
+              if (klass.isAnnotation()
+                  && input.isAnnotationPresent((Class<? extends Annotation>) klass)) {
+                return true;
+              }
+            }
+            return false;
+          }
+        },
+        Default.class);
+
+    private final Predicate<Method> predicate;
+    private final Class<? extends Annotation> annotationClass;
+
+    AnnotationPredicate(Predicate<Method> predicate, Class<? extends Annotation> annotationClass) {
```

Comment:
Are you sure?, your currently not using composition within the AnnotationPredicates class.